src/components/pages/profile/[name]/registration/steps/Pricing/Pricing.tsx

```diff
@@ -524,43 +529,45 @@ const Pricing = ({
     registrationData: {
       ...registrationData,
       reverseRecord,
-      years,
+      seconds,
       records: [{ key: 'ETH', value: resolverAddress, type: 'addr', group: 'address' }],
       clearRecords: resolverExists,
       resolverAddress,
     },
   })
-  const { hasPremium, premiumFee, gasPrice, yearlyFee, totalYearlyFee, estimatedGasFee } =
+
+  const { hasPremium, premiumFee, gasPrice, yearlyFee, totalDurationBasedFee, estimatedGasFee } =
     fullEstimate
+  const durationRequiredBalance = totalDurationBasedFee ? (totalDurationBasedFee * 110n) / 100n : 0n
+  const totalRequiredBalance = durationRequiredBalance
+    ? durationRequiredBalance + (premiumFee || 0n) + (estimatedGasFee || 0n)
+    : 0n
+
+  const previousYearlyFee = usePreviousDistinct(yearlyFee) || 0n
 
```

Comment:
What was the issue for why usePreviousDistinct was needed?

Reply:
the gas preview was incorrect